ui/src/plugins/dev.perfetto.ExplorePage/query_builder/builder.ts

```diff
@@ -450,12 +489,54 @@ export class Builder implements m.ClassComponent<BuilderAttrs> {
                 this.selectedView = SelectedView.kComment;
                 this.isExplorerCollapsed = false;
               }
+              this.savePanelState();
             },
           }),
         ),
     );
   }
 
+  private loadPanelState(): PanelState {
```

Comment:
I have few issues with this:

- We should use a central place for storing all workspace layout configuration going forward.
- 2 separate entries seems a little excessive.
- The type assertion isn't type safe at all - use Zod if you're parsing objects of unknown types.
- I think keeping the panel state consistent for the duration of the session is perfectly fine for now.